subsys/bluetooth/host/classic/ssp.c

```diff
@@ -650,6 +650,19 @@ void bt_hci_io_capa_req(struct net_buf *buf)
 		return;
 	}
 
+#if defined(CONFIG_BT_SMP_APP_PAIRING_ACCEPT)
+	if (bt_auth && bt_auth->pairing_accept) {
+		enum bt_security_err err;
+
+		err = bt_auth->pairing_accept(conn, NULL);
+		if (err != BT_SECURITY_ERR_SUCCESS) {
+			io_capa_neg_reply(&evt->bdaddr,
+					  BT_HCI_ERR_PAIRING_NOT_ALLOWED);
+			return;
```

Comment:
Not familiar with legacy pairing, but it seems odd to use `pairing_accept` for the IO capability request. 

`pairing_accept` should be called when we receive the initial pairing request; is this the first request that is sent for legacy pairing?

Reply:
> pairing_accept should be called when we receive the initial pairing request; is this the first request that is sent for legacy pairing?

I believe so, yes, but I'll let @MarkWangChinese confirm. You could think of BR/EDR pairing as if SMP was running on the controller side, e.g. the host will not see the initial communication that happens between peers (such as the equivalent of "initial pairing request").

Reply:
> You could think of BR/EDR pairing as if SMP was running on the controller side, e.g. the host will not see the initial communication that happens between peers (such as the equivalent of "initial pairing request").

The first point where the user can intervene is the BT_HCI_EVT_IO_CAPA_REQ event.

Reply:
Vol 2, Part C, 4.2.7 shows the SSP's four stages as:
- IO capabilities exchange
- Public key exchange
- Authentication stage 1
- Authentication stage 2

Vol 2, Part F, 4.2.6 shows the HCI related cmd for SSP's start.
Vol 2, Part F, 4.2.7 shows the HCI related cmd of IO capability exchange. Yes, the air packet is sent by controller, it is not SMP packet. it is LMP packet.
Vol 4, Part E, 7.1.36 describes the "HCI_IO_Capability_Request_Negative_Reply" as: The HCI_IO_Capability_Request_Negative_Reply command shall be used to
reject a pairing attempt after an HCI_IO_Capability_Request event has been received by the Host.

I tried the change with Android phone. If pairing_accept return error to send HCI_IO_Capability_Request_Negative_Reply, the Android phone side show information like "Pairing rejected by xxx".